File: ApiCall/OllamaAPI.py
import os
import sys
import json
import requests
from datetime import datetime

# Default parameters (can be overridden by environment variables or CLI args)
hostApi = os.environ.get("HOST_API", "http://localhost:11434")
model = os.environ.get("MODEL", "phi3:mini")
prompt = os.environ.get("PROMPT", "What is the capital of France?")
printRaw = os.environ.get("PRINT_RAW", "false").lower() == "true"
enableLogging = os.environ.get("ENABLE_LOGGING", "false").lower() == "true"

# Parse CLI args (optional, simple)
for arg in sys.argv[1:]:
    if arg.startswith("--hostApi="): hostApi = arg.split("=",1)[1]
    elif arg.startswith("--model="): model = arg.split("=",1)[1]
    elif arg.startswith("--prompt="): prompt = arg.split("=",1)[1]
    elif arg == "--printRaw": printRaw = True
    elif arg == "--enableLogging": enableLogging = True

logFile = os.path.join(os.path.dirname(__file__), "llm.log")

def write_log(message):
    timestamp = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
    with open(logFile, "a", encoding="utf-8") as f:
        f.write(f"{timestamp} {message}\n")

if enableLogging:
    write_log(f"Run: hostApi='{hostApi}', model='{model}', prompt='{prompt}', printRaw={printRaw}")

body = {"model": model, "prompt": prompt}

try:
    resp = requests.post(f"{hostApi}/api/generate", json=body)
    if enableLogging:
        write_log(f"STATUS CODE: {resp.status_code}")
    print(f"STATUS CODE: {resp.status_code}")
    responseString = resp.text
    fullResponse = ""
    for line in responseString.splitlines():
        if line.strip():
            try:
                j = json.loads(line)
                if "response" in j:
                    fullResponse += j["response"]
            except Exception:
                pass
    if printRaw:
        print(responseString)
        if enableLogging:
            write_log(f"RAW RESPONSE: {responseString[:200]}...")
    else:
        print(fullResponse)
        if enableLogging:
            write_log(f"FULL RESPONSE: {fullResponse[:200]}...")
except Exception as e:
    print("ERROR:")
    print(e)
    if enableLogging:
        write_log(f"ERROR: {e}")

File: ApiCall/OllamaAPI.cs
using System;
using System.IO;
using System.Net.Http;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;

class OllamaAPI
{
    static async Task Main(string[] args)
    {
        // Default parameters
        string hostApi = "http://localhost:11434";
        string model = "phi3:mini";
        string prompt = "What is the capital of France?";
        bool printRaw = false;
        bool enableLogging = false;

        // Parse command-line arguments (optional)
        foreach (var arg in args)
        {
            if (arg.StartsWith("-hostApi=")) hostApi = arg.Substring(9);
            else if (arg.StartsWith("-model=")) model = arg.Substring(7);
            else if (arg.StartsWith("-prompt=")) prompt = arg.Substring(8);
            else if (arg == "-printRaw") printRaw = true;
            else if (arg == "-enableLogging") enableLogging = true;
        }

        string logFile = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "llm.log");
        void WriteLog(string message)
        {
            string timestamp = DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss");
            File.AppendAllText(logFile, $"{timestamp} {message}{Environment.NewLine}");
        }

        if (enableLogging)
            WriteLog($"Run: hostApi='{hostApi}', model='{model}', prompt='{prompt}', printRaw={printRaw}");

        var bodyObj = new { model, prompt };
        string body = JsonSerializer.Serialize(bodyObj);

        try
        {
            using var client = new HttpClient();
            var content = new StringContent(body, Encoding.UTF8, "application/json");
            var response = await client.PostAsync($"{hostApi}/api/generate", content);
            string responseString = await response.Content.ReadAsStringAsync();

            if (enableLogging)
                WriteLog($"STATUS CODE: {(int)response.StatusCode}");

            Console.WriteLine($"STATUS CODE: {(int)response.StatusCode}");

            var fullResponse = new StringBuilder();
            var lines = responseString.Split('\n');
            foreach (var line in lines)
            {
                if (!string.IsNullOrWhiteSpace(line))
                {
                    try
                    {
                        using var doc = JsonDocument.Parse(line);
                        if (doc.RootElement.TryGetProperty("response", out var respProp))
                            fullResponse.Append(respProp.GetString());
                    }
                    catch { }
                }
            }

            if (printRaw)
            {
                Console.WriteLine(responseString);
                if (enableLogging)
                    WriteLog($"RAW RESPONSE: {responseString.Substring(0, Math.Min(200, responseString.Length))}...");
            }
            else
            {
                Console.WriteLine(fullResponse.ToString());
                if (enableLogging)
                    WriteLog($"FULL RESPONSE: {fullResponse.ToString().Substring(0, Math.Min(200, fullResponse.Length))}...");
            }
        }
        catch (Exception ex)
        {
            Console.WriteLine("ERROR:");
            Console.WriteLine(ex);
            if (enableLogging)
                WriteLog($"ERROR: {ex}");
        }
    }
}